consensus/istanbul/core/events.go

```diff
@@ -30,3 +30,8 @@ type resendRoundChangeEvent struct {
 type timeoutAndMoveToNextRoundEvent struct {
 	view *istanbul.View
 }
+
+// NewViewEvent is posted when a new view has been created
+type NewViewEvent struct {
+	View *istanbul.View
```

Comment:
You may need to add a field that contains the new proposer to this event.  Especially if you need to emit this event to the worker for all new views.